mesonbuild/modules/pkgconfig.py

```diff
@@ -374,6 +375,15 @@ def _generate_pkgconfig_file(self, state, deps, subdirs, name, description,
         else:
             outdir = state.environment.scratch_dir
             prefix = PurePath(coredata.get_option(mesonlib.OptionKey('prefix')))
+            if pkgroot:
+                pkgroot = PurePath(pkgroot)
+                if not pkgroot.is_absolute():
+                    pkgroot = prefix / pkgroot
+                elif prefix not in pkgroot.parents:
+                    raise mesonlib.MesonException('Pkgconfig prefix cannot be outside of the prefix '
+                                                  'when pkgconfig.relocatable=true. '
+                                                  f'Pkgconfig prefix is {pkgroot}.')
```

Comment:
This currently prints an error in CI pointing at either:
```
C:\opt\lib\pkgconfig
```
or
```
/opt/lib/pkgconfig
```

This is challenging to detect in a regex match in test.json, because you need to change lots of things. You can simplify it quite a bit by printing out an error message of the form 
```suggestion
                                                  f'Pkgconfig prefix is {pkgroot.as_posix()}.')
```

Reply:
Thanks! I was going a bit insane trying to escape `\` for the path in the `test.json`. Changed it to transform the path into posix in the error message.

Hopefully if will go through :crossed_fingers:

Reply:
Nope :disappointed:, looks like I somehow need to deal with the drive letter. I'll see if I get some time working on it the following days otherwise I'll continue on Monday.

Reply:
Yes, your previous regex approach seems reasonable there.

What as_posix() does is ensures you don't need to *also* handle the difference between \ and / and particularly getting the former to be correctly json-escaped. Handling `C:` via regex is comparatively trivial.

Reply:
Ah, ok. I thought there was more than just the `\` and `/` when using the regex. Rolling back to using that with the `as_posix()`. We'll see how that goes.